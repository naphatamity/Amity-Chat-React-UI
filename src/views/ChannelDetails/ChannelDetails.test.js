import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { ChannelDetails } from '.';

const channelId = 'mock-channel-id';

describe('<ChannelDetails /> spec', () => {
  it('renders the component', () => {
    const container = render(
      <Router>
        <ChannelDetails channelId={channelId} />
      </Router>,
    );
    expect(container).toMatchSnapshot();
  });
});
