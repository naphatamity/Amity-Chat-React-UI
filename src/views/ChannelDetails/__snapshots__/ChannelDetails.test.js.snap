// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChannelDetails /> spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdvvaa cQfXqj"
      >
        <div
          class="sc-iCfLBT ehJiFL"
        >
          <a
            href="/channel/mock-channel-id"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-chevron-left fa-w-8 "
              data-icon="chevron-left"
              data-prefix="far"
              focusable="false"
              role="img"
              style="margin-right: 8px;"
              viewBox="0 0 256 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M231.293 473.899l19.799-19.799c4.686-4.686 4.686-12.284 0-16.971L70.393 256 251.092 74.87c4.686-4.686 4.686-12.284 0-16.971L231.293 38.1c-4.686-4.686-12.284-4.686-16.971 0L4.908 247.515c-4.686 4.686-4.686 12.284 0 16.971L214.322 473.9c4.687 4.686 12.285 4.686 16.971-.001z"
                fill="currentColor"
              />
            </svg>
          </a>
          <div
            class="sc-jRQAMF jtZgMJ"
          >
            <h2
              class="sc-gKckTs DFHsz"
            >
              Channel Details
            </h2>
          </div>
        </div>
        <div
          class="sc-gsDJrp eZKoEd"
        >
          <ul
            class="sc-dkPtyc wnnbT"
          >
            <li
              class="sc-hKwCoD gmiScZ"
            >
              <div>
                Report User
              </div>
            </li>
            <li
              class="sc-hKwCoD gmiScZ"
            >
              <div
                class="alert"
              >
                Leave Chat
              </div>
            </li>
            <li
              class="sc-hKwCoD gmiScZ"
            >
              <div>
                <div
                  class="sc-eCImvq iYsfFg"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-pen fa-w-16 "
                    data-icon="pen"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                Group Profile
              </div>
              <div>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-chevron-right fa-w-8 "
                  data-icon="chevron-right"
                  data-prefix="far"
                  focusable="false"
                  role="img"
                  viewBox="0 0 256 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M24.707 38.101L4.908 57.899c-4.686 4.686-4.686 12.284 0 16.971L185.607 256 4.908 437.13c-4.686 4.686-4.686 12.284 0 16.971L24.707 473.9c4.686 4.686 12.284 4.686 16.971 0l209.414-209.414c4.686-4.686 4.686-12.284 0-16.971L41.678 38.101c-4.687-4.687-12.285-4.687-16.971 0z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </li>
            <li
              class="sc-hKwCoD gmiScZ"
            >
              <div>
                <div
                  class="sc-eCImvq iYsfFg"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-user-friends fa-w-20 "
                    data-icon="user-friends"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 640 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M192 256c61.9 0 112-50.1 112-112S253.9 32 192 32 80 82.1 80 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C51.6 288 0 339.6 0 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zM480 256c53 0 96-43 96-96s-43-96-96-96-96 43-96 96 43 96 96 96zm48 32h-3.8c-13.9 4.8-28.6 8-44.2 8s-30.3-3.2-44.2-8H432c-20.4 0-39.2 5.9-55.7 15.4 24.4 26.3 39.7 61.2 39.7 99.8v38.4c0 2.2-.5 4.3-.6 6.4H592c26.5 0 48-21.5 48-48 0-61.9-50.1-112-112-112z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                Members
              </div>
              <div>
                <span>
                  5
                </span>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-chevron-right fa-w-8 "
                  data-icon="chevron-right"
                  data-prefix="far"
                  focusable="false"
                  role="img"
                  style="margin-left: 8px;"
                  viewBox="0 0 256 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M24.707 38.101L4.908 57.899c-4.686 4.686-4.686 12.284 0 16.971L185.607 256 4.908 437.13c-4.686 4.686-4.686 12.284 0 16.971L24.707 473.9c4.686 4.686 12.284 4.686 16.971 0l209.414-209.414c4.686-4.686 4.686-12.284 0-16.971L41.678 38.101c-4.687-4.687-12.285-4.687-16.971 0z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdvvaa cQfXqj"
    >
      <div
        class="sc-iCfLBT ehJiFL"
      >
        <a
          href="/channel/mock-channel-id"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chevron-left fa-w-8 "
            data-icon="chevron-left"
            data-prefix="far"
            focusable="false"
            role="img"
            style="margin-right: 8px;"
            viewBox="0 0 256 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M231.293 473.899l19.799-19.799c4.686-4.686 4.686-12.284 0-16.971L70.393 256 251.092 74.87c4.686-4.686 4.686-12.284 0-16.971L231.293 38.1c-4.686-4.686-12.284-4.686-16.971 0L4.908 247.515c-4.686 4.686-4.686 12.284 0 16.971L214.322 473.9c4.687 4.686 12.285 4.686 16.971-.001z"
              fill="currentColor"
            />
          </svg>
        </a>
        <div
          class="sc-jRQAMF jtZgMJ"
        >
          <h2
            class="sc-gKckTs DFHsz"
          >
            Channel Details
          </h2>
        </div>
      </div>
      <div
        class="sc-gsDJrp eZKoEd"
      >
        <ul
          class="sc-dkPtyc wnnbT"
        >
          <li
            class="sc-hKwCoD gmiScZ"
          >
            <div>
              Report User
            </div>
          </li>
          <li
            class="sc-hKwCoD gmiScZ"
          >
            <div
              class="alert"
            >
              Leave Chat
            </div>
          </li>
          <li
            class="sc-hKwCoD gmiScZ"
          >
            <div>
              <div
                class="sc-eCImvq iYsfFg"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-pen fa-w-16 "
                  data-icon="pen"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              Group Profile
            </div>
            <div>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-right fa-w-8 "
                data-icon="chevron-right"
                data-prefix="far"
                focusable="false"
                role="img"
                viewBox="0 0 256 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M24.707 38.101L4.908 57.899c-4.686 4.686-4.686 12.284 0 16.971L185.607 256 4.908 437.13c-4.686 4.686-4.686 12.284 0 16.971L24.707 473.9c4.686 4.686 12.284 4.686 16.971 0l209.414-209.414c4.686-4.686 4.686-12.284 0-16.971L41.678 38.101c-4.687-4.687-12.285-4.687-16.971 0z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </li>
          <li
            class="sc-hKwCoD gmiScZ"
          >
            <div>
              <div
                class="sc-eCImvq iYsfFg"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-user-friends fa-w-20 "
                  data-icon="user-friends"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 640 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M192 256c61.9 0 112-50.1 112-112S253.9 32 192 32 80 82.1 80 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C51.6 288 0 339.6 0 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zM480 256c53 0 96-43 96-96s-43-96-96-96-96 43-96 96 43 96 96 96zm48 32h-3.8c-13.9 4.8-28.6 8-44.2 8s-30.3-3.2-44.2-8H432c-20.4 0-39.2 5.9-55.7 15.4 24.4 26.3 39.7 61.2 39.7 99.8v38.4c0 2.2-.5 4.3-.6 6.4H592c26.5 0 48-21.5 48-48 0-61.9-50.1-112-112-112z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              Members
            </div>
            <div>
              <span>
                5
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-right fa-w-8 "
                data-icon="chevron-right"
                data-prefix="far"
                focusable="false"
                role="img"
                style="margin-left: 8px;"
                viewBox="0 0 256 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M24.707 38.101L4.908 57.899c-4.686 4.686-4.686 12.284 0 16.971L185.607 256 4.908 437.13c-4.686 4.686-4.686 12.284 0 16.971L24.707 473.9c4.686 4.686 12.284 4.686 16.971 0l209.414-209.414c4.686-4.686 4.686-12.284 0-16.971L41.678 38.101c-4.687-4.687-12.285-4.687-16.971 0z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
